### ISSUE TITLE
Deprecation warning with Sundials version 6.0.0

### ISSUE THREAD

**Issue by ischoegl** [Cantera/cantera#1349]
**Problem description**

Compilation with Sundials 6.0 results in the following warnings:
```
src/numerics/BandMatrix.cpp: In member function 'virtual int Cantera::BandMatrix::factor()':
src/numerics/BandMatrix.cpp:247:23: warning: 'sunindextype bandGBTRF(realtype**, sunindextype, sunindextype, sunindextype, sunindextype, sunindextype*)' is deprecated: use SUNDlsMat_bandGBTRF instead [-Wdeprecated-declarations]
  247 |     m_info = bandGBTRF(m_lu_col_ptrs.data(), static_cast<long int>(nColumns()),
      |              ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  248 |                        nu, nl, smu, m_ipiv->data.data());
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/ischoegl/.conda/envs/cantera-dev/include/sunlinsol/sunlinsol_band.h:30,
                 from src/numerics/BandMatrix.cpp:13:
/home/ischoegl/.conda/envs/cantera-dev/include/sundials/sundials_band.h:91:14: note: declared here
   91 | sunindextype bandGBTRF(realtype **a, sunindextype n,
      |              ^~~~~~~~~
src/numerics/BandMatrix.cpp: In member function 'virtual int Cantera::BandMatrix::solve(doublereal*, size_t, size_t)':
src/numerics/BandMatrix.cpp:281:14: warning: 'void bandGBTRS(realtype**, sunindextype, sunindextype, sunindextype, sunindextype*, realtype*)' is deprecated: use SUNDlsMat_bandGBTRS instead [-Wdeprecated-declarations]
  281 |     bandGBTRS(a, static_cast<long int>(nColumns()), smu, nl, m_ipiv->data.data(), b);
      |     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/ischoegl/.conda/envs/cantera-dev/include/sunlinsol/sunlinsol_band.h:30,
                 from src/numerics/BandMatrix.cpp:13:
/home/ischoegl/.conda/envs/cantera-dev/include/sundials/sundials_band.h:124:6: note: declared here
  124 | void bandGBTRS(realtype **a, sunindextype n, sunindextype smu,
      |      ^~~~~~~~~

```

**Steps to reproduce**

1. Run `scons build`

**Behavior**

**System information**

- Cantera version: 3.0.0a2
- Sundials: 6.0.0

**Comment by bryanwweber** [comment]
@ischoegl Is this resolved by #1346?

**Comment by ischoegl** [comment]
> @ischoegl Is this resolved by #1346?

Unfortunately not.
